src/compiler/commandLineParser.ts

```diff
@@ -709,14 +713,16 @@ namespace ts {
         options.configFilePath = configFileName;
 
         const { fileNames, wildcardDirectories } = getFileNames(errors);
+        const compileOnSave = <boolean>convertCompileOnSaveOptionFromJson(json, basePath, errors);

```

Comment:
make `convertCompileOnSaveOptionFromJson` return `boolean` to remove type assertion